pyschema/source_generation.py

```diff
@@ -78,48 +183,50 @@ def _class_source(schema, indent):
 
 class CachedGraphTraverser(object):
     def __init__(self):
-        self.descendents = {}
+        self.descendants = {}
         self.started = set()
 
-    def find_descendents(self, a):
-        if a in self.descendents:
+    def find_descendants(self, a, max_depth=sys.getrecursionlimit()):
```

Comment:
But it can still hit the recursion limit right? Since any "field" traversal (as opposed to record traversal) won't decrease the max_depth variable but still deepen the recursion depth... Anyway, no biggie. Might refactor this at some later point to roll up the recursion into a BFS-style loop